source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -361,6 +361,22 @@ std::string SslSocket::subjectLocalCertificate() const {
   return Utility::getSubjectFromCertificate(*cert);
 }
 
+absl::optional<SystemTime> SslSocket::validFromPeerCertificate() const {
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl_.get()));
+  if (!cert) {
+    return absl::optional<SystemTime>();
```

Comment:
nit: `absl::nullopt`. Also do you have test coverage for this path here and below?